### PR TITLE
Fix 53003 on room invite sms shown

### PR DESCRIPTION
**Pull request by Kalydosos**
### Fixed Issues

$ https://github.com/Expensify/App/issues/53003
PROPOSAL: https://github.com/Expensify/App/issues/53003#issuecomment-2495946266

### Tests

1. Launch app
2. Open a room chat
3. Invite a phone number via suggestion eg: +919789945670
4. Send the message
5. From whisper, tap invite
6. Tap header -- member
7. Open the phone number member
8. Note expensify.sms is not shown with phone number

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/27f24634-140f-4e3a-9cf7-de6f835d3f62
</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f46e0e21-5fe4-4d67-b3cc-9d8ca3464ea6
</details>

<details>
<summary>iOS: Native</summary>

ios_native

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios_mweb_safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

fix_ios_safari
</details>

<details>
<summary>MacOS: Desktop</summary>

macos-desktop

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
Thanks for the PR! Please also address the ESLint errors and fix the unit tests.

**Comment by Kalydosos** [comment]
ok, i'm working on it

**Comment by Kalydosos** [comment]
> Thanks for the PR! Please also address the ESLint errors and fix the unit tests.

the unit tests said failed seems have passed but they are marked failed, what do you think ?

**Comment by jayeshmangwani** [comment]
> the unit tests said failed seems have passed but they are marked failed, what do you think ?

maybe try to merge latest main branch and push again

**Comment by jayeshmangwani** [comment]
@Kalydosos Have you tested all 13 files where you made these changes?

**Comment by jayeshmangwani** [comment]
@Kalydosos Lint is failing here, can you please fix it

**Comment by jayeshmangwani** [comment]
> the problem with the tests is the 'NO-BREAK SPACE' (U+00A0) character used here

I believe we shouldn't change the tests in this PR, as the failures don't seem to be caused by our changes. Or am I missing something?

**Comment by Kalydosos** [comment]
> I believe we shouldn't change the tests in this PR, as the failures don't seem to be caused by our changes. Or am I missing something?

@jayeshmangwani formatPhoneNumber replace simple space characters by no-break space characters (which make sense)
need_change_tests

**Comment by Kalydosos** [comment]
> @Kalydosos Lint is failing here, can you please fix it

@jayeshmangwani the current lint failure have to do with the legacy use of withOnyx in some of the files. It is not from our changes. I guess to deploy on staging they will have to skip that test https://github.com/Expensify/App/actions/runs/12071160259/job/33662272975?pr=53240

File: src/pages/RoomMemberDetailsPage.tsx [hunk 1]
```diff
@@ -47,7 +48,7 @@ function RoomMemberDetailsPage({report, route}: RoomMemberDetailsPagePageProps)
     const member = report?.participants?.[accountID];
     const details = personalDetails?.[accountID] ?? ({} as PersonalDetails);
     const fallbackIcon = details.fallbackIcon ?? '';
-    const displayName = details.displayName ?? '';
+    const displayName = formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details));
```
[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @jayeshmangwani i'll check that

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: Haha, no problem! Keeping discussion in one place makes things faster, so 😃 .

[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: no it's needed @jayeshmangwani i did a test again
Capture d’écran de 2024-12-02 18-08-41

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @Kalydosos I am adding the steps below; please try them on your side as well.

1. Add a user with a phone number.
2. Remove the `formatPhoneNumber` function from `RoomMemberDetailsPage`.

We can observe(console log) that the result of `PersonalDetailsUtils.getDisplayNameOrDefault(details)` is the same as `formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details))`. This indicates that the formatPhoneNumber function is redundant in this context. Since we are already formatting the phone number, it doesn't need to be included here.

I’ve attached a video for reference. If anything is still unclear, feel free to let me know.

https://github.com/user-attachments/assets/cf35d0bd-ae44-484d-8be9-db445cec7f4d

File: src/libs/OptionsListUtils.ts [hunk 2]
```diff
@@ -1333,20 +1333,20 @@ function getShareLogOptions(options: OptionList, betas: Beta[] = []): Options {
...
                 source: personalDetail?.avatar ?? FallbackAvatar,
-                name: personalDetail?.login ?? '',
+                name: LocalePhoneNumber.formatPhoneNumber(personalDetail?.login ?? ''),
                 type: CONST.ICON_TYPE_AVATAR,
                 id: personalDetail?.accountID,
             },
         ],
         descriptiveText: amountText ?? '',
-        login: personalDetail?.login ?? '',
+        login: LocalePhoneNumber.formatPhoneNumber(personalDetail?.login ?? ''),
```
Comment (jayeshmangwani) [inline src/libs/OptionsListUtils.ts:1349]: Please avoid updating the login here. Instead, we will only update the `displayName` where it is used to ensure that we do not introduce any regressions.

File: src/libs/OptionsListUtils.ts [hunk 3]
```diff
@@ -1333,20 +1333,20 @@ function getShareLogOptions(options: OptionList, betas: Beta[] = []): Options {
...
     return {
-        text: PersonalDetailsUtils.getDisplayNameOrDefault(personalDetail, formattedLogin),
-        alternateText: formattedLogin || PersonalDetailsUtils.getDisplayNameOrDefault(personalDetail, '', false),
+        text: LocalePhoneNumber.formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(personalDetail, login)),
+        alternateText: LocalePhoneNumber.formatPhoneNumber(login || PersonalDetailsUtils.getDisplayNameOrDefault(personalDetail, '', false)),
         icons: [
             {
                 source: personalDetail?.avatar ?? FallbackAvatar,
-                name: personalDetail?.login ?? '',
+                name: LocalePhoneNumber.formatPhoneNumber(personalDetail?.login ?? ''),
```
Comment (jayeshmangwani) [inline src/libs/OptionsListUtils.ts:1343]: Is this change necessary here? Instead, we can simply use `LocalePhoneNumber.formatPhoneNumber` directly where the `icons.name` key is being used.

File: src/pages/RoomMemberDetailsPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @jayeshmangwani maybe you had a usage of these accounts which end up setting the PersonalDetailsUtils.getDisplayNameOrDefault(details) as same value with the formatPhoneNumber value, as i have a different result as you can see. A good test will be to invite a new user with a new phone number. But anyway, you agree with me that we dont expect PersonalDetailsUtils.getDisplayNameOrDefault to act the same as formatPhoneNumber right ? And as using formatPhoneNumber seems to not harm the end result in both our cases i think we should keep it as not using it have the inverse impact in my test case. I'm saying our test cases  show that it's better to have it there instead of not to have it, as in my case the result is unsatisfying (the core issue of this ticket). What do you think ?

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: > What do you think ?

On this PR `formatPhoneNumber` change is being used in many places throughout. If even one instance where `formatPhoneNumber` is not needed is included, it could lead to unnecessary regressions.

What I suggest is that we only use `formatPhoneNumber` in this PR at locations where the changes can be tested easily, and we can verify the outcome of those specific changes. Additionally, the two files `src/libs/ReportUtils.ts` and `src/libs/OptionsListUtils.ts` are used in several places. Any changes to these files should be made more cautiously to avoid potential issues.

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: > you agree with me that we dont expect PersonalDetailsUtils.getDisplayNameOrDefault to act the same as formatPhoneNumber right ?

Agreed

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: > And as using formatPhoneNumber seems to not harm the end result in both our cases i think we should keep it as not using it have the inverse impact in my test case

I don't agree with this point. My thinking is that if the text is already formatted the way we want, there's no need to format it again. Reapplying formatting might introduce issues or break functionality at some point, which we can't predict.

[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @jayeshmangwani this is a test without formatPhoneNumber, as you can see, on the previous screens fromatting is correct but on roommemberdetails screen, even though +22951323930 is in the same region as me its details appear unformatted the same manner as +27440370877 which is in a different region. What do you suggess ? 

https://github.com/user-attachments/assets/0d809a82-a870-4445-842d-be03ebbf2425

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: > What do you suggess ?
@Kalydosos Sorry, I’m a little confused— is this video from your branch or the main branch?

[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: it's my branch @jayeshmangwani

[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: i will do a new merge with main

[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @jayeshmangwani i did a test again after the new merge and it's the result result as in the video

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @Kalydosos For me, it’s working fine both with and without using the `formatPhoneNumber`. I’m not sure why we’re getting different results.

[the remark this answers is not in the document]
Comment (jayeshmangwani) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @Kalydosos Please leave the `RoomMemberDetailsPage.tsx` page as it is with the changes you’ve made; we’ll figure this out later. However, can you please check the other comments? I feel we shouldn’t make changes in a few places that I’ve highlighted.

File: src/libs/OptionsListUtils.ts (hunk 2 above)
Comment (Kalydosos) [reply src/libs/OptionsListUtils.ts:1349]: ok, i did a test on spliting expenses and i agree. It will be good if you can also confirm by a visual test

File: src/libs/OptionsListUtils.ts (hunk 3 above)
Comment (Kalydosos) [reply src/libs/OptionsListUtils.ts:1343]: same here, i did a test on spliting expenses and i agree. It will be good if you can also confirm by a visual test

**Comment by Kalydosos** [comment]
some tests failed locally but i think it's a due to local env

**Comment by Kalydosos** [comment]
@jayeshmangwani what's next ?

**Comment by jayeshmangwani** [comment]
> what's next ?

@Kalydosos I have tested the code, and there are a few things that I think we need to address:

1. We need to migrate the usage from withOnyx to useOnyx on the ReportParticipantDetails and ArchivedReportFooter pages.
2. I am unable to assign you the cards. If we cannot test that change right now, we should revert it.
3. Apart from the cards pages, have you tested all the other pages where you made changes?

**Comment by Kalydosos** [comment]
@jayeshmangwani there are 3 files (also ProfileAvatar) with withOnyx, that's a lot of potential regressions, really i'm not a fan of the idea, considering also the widespread of our changes. Cant you test the cards otherwise i'll revert these (i dont think phone numbers are displayed on cards anyway). I have tested all the other pages.

**Comment by jayeshmangwani** [comment]
> that's a lot of potential regressions

@Kalydosos What are you seeing that’s very hard to test in those files, and could potentially cause regression?

**Comment by Kalydosos** [comment]
i will make the migrations @jayeshmangwani, but there is often some unexpected bug when fixing many things at once

**Comment by jayeshmangwani** [comment]
Yes that may be a chance but let's try to test it thoroughly

**Comment by Kalydosos** [comment]
@jayeshmangwani i did the migrations and tested it. All seems right but i would appreciate if you can do a visual check on ArchivedReportFooter and confirms the behavior as correct. Thx. If its all good i think we are all set then for merging. Agreed ?

**Comment by jayeshmangwani** [comment]
Thank you @Kalydosos for implementing the `useOnyx` changes.

I’ve visually tested the updates to ArchivedReportFooter and have some feedback to share. Additionally, I’ve noticed a few other issues. Let me complete testing on other pages first, and then I’ll post a consolidated review.

**Comment by jayeshmangwani** [comment]
@Kalydosos If you want to test the `ArchivedReportFooter`, follow these steps:

1. Go to any workspace.
2. Invite a user with a phone number.
3. Remove that member from the workspace.
4. Navigate to the home page, then search for that user (by their phone number) and the corresponding workspace chat.
5. Check the message displayed at the bottom.

Based on my testing in the staging environment, it seems we don’t need to make any changes to the `ArchivedReportFooter`. The formatted phone number is already displayed correctly. However, let me know if your results differ from mine.
Screenshot 2024-12-05 at 22 48 29

**Comment by jayeshmangwani** [comment]
We haven’t added the changes to the Workspace Member Details page yet, so we need to update that page as well.

Steps to Verify:

1. Go to any workspace.
2. Invite a user with a phone number.
3. Click on that user to open their details.
4. Check the header and the phone number displayed. You’ll notice it currently shows ".sms".
Screenshot 2024-12-05 at 22 47 31

**Comment by Kalydosos** [comment]
@jayeshmangwani about ArchivedReportFooter i could not found the user (step 4). Maybe you can show me that step in video. And about that file, we dont modify the display names, we format it when displaying the reason of the archiving but it seems like that reason is not displayed anymore. Anyway, i remove the change as i couldnt see the reason displayed (i just make sure the page is displayed with no obvious bug)
https://github.com/Kalydosos/App/blob/fix-53003-on-room-invite-sms-shown/src/components/ArchivedReportFooter.tsx#L53-L61

In the process, i fix a few other places where i see the bug including the Workspace Member Details page
workspacemember_before
workspacemember_after

**Comment by jayeshmangwani** [comment]
@Kalydosos I hope you have tested other pages where the username can include .sms

**Comment by Kalydosos** [comment]
> @Kalydosos I hope you have tested other pages where the username can include .sms

the ones i modified yes

**Comment by jayeshmangwani** [comment]
> the ones i modified yes

Not where you have modified, but rather where we are using the `displayName`

**Comment by jayeshmangwani** [comment]
> about ArchivedReportFooter i could not found the user (step 4). Maybe you can show me that step in video

I hope the video below will help, but let me know if these steps don't work.

https://github.com/user-attachments/assets/a823ea6c-702c-42c7-a880-d34a13627a5e

**Comment by Kalydosos** [comment]
@jayeshmangwani i dont understand clearly what you are meaning. There a lot of pages where we display sms in the display name as i have notice, that should be a whole refactoring ticket as it involves also librairies like Search. We cant not exhaustively get it all done in this ticket here. A ticket should be specific for it, with a bounty that matches that extensive effort. We have done a lot of changes in this ticket already, we can not reasonably do everything in this one ticket

**Comment by jayeshmangwani** [comment]
@Kalydosos I'm not sure what to say here, but at the very least, we can address the issue we’ve identified since it involves only a single-line change. The workspace member details were a clear and manageable change that we could have handled earlier. I feel a simple change has been stretched unnecessarily. Please fix any issues you encounter during testing, and I’ll report if I find anything after your changes. It’s okay; we don’t need to refactor everything at this stage.

cc: @luacmartins

**Comment by Kalydosos** [comment]
@jayeshmangwani i did. But there are surely other pages out there with same issue. I cannot start tracking them in the context of this ticet that's what i'm saying. I can eventually start afresh in a new ticket dedicated to that objective, the process will then be different and the result assured. we are indeed overstretching here and it's confusing as one can easily loose focus.

**Comment by Kalydosos** [comment]
> > about ArchivedReportFooter i could not found the user (step 4). Maybe you can show me that step in video
> 
> I hope the video below will help, but let me know if these steps don't work.

yes i remove it already. And the reason dont display anymore it seems.

**Comment by jayeshmangwani** [comment]
> And the reason dont display anymore it seems.

Sorry, what dont display ?

**Comment by jayeshmangwani** [comment]
> @jayeshmangwani i did. But there are surely other pages out there with same issue. I cannot start tracking them in the context of this ticet that's what i'm saying. I can eventually start afresh in a new ticket dedicated to that objective, the process will then be different and the result assured. we are indeed overstretching here and it's confusing as one can easily loose focus.

No issues, @Kalydosos . In this PR, we are addressing only the page where the original issue was raised, along with a few other pages where the same issue was identified during testing.

@luacmartins, I’m tagging you here to get your consensus. Please note that in this PR, we are not handling the entire app's usage of displayName. There might still be some pages where users can see the .sms in the phone number.

**Comment by Kalydosos** [comment]
> > And the reason dont display anymore it seems.
> 
> Sorry, what dont display ?


https://github.com/Kalydosos/App/blob/fix-53003-on-room-invite-sms-shown/src/components/ArchivedReportFooter.tsx#L53-L61

**Comment by Kalydosos** [comment]
> No issues, @Kalydosos . In this PR, we are addressing only the page where the original issue was raised, along with a few other pages where the same issue was identified during testing.

i agree

> there might still be some pages where users can see the .sms in the phone number.

i agree

File: src/components/ArchivedReportFooter.tsx [hunk 4]
```diff
@@ -60,7 +53,7 @@ function ArchivedReportFooter({report, reportClosedAction, personalDetails = {}}
     const text = shouldRenderHTML
         ? translate(`reportArchiveReasons.${archiveReason}`, {
               displayName: `<strong>${displayName}</strong>`,
-              oldDisplayName: `<strong>${oldDisplayName}</strong>`,
+              oldDisplayName: `<strong>${oldDisplayName ?? ''}</strong>`,
```
Comment (jayeshmangwani) [inline src/components/ArchivedReportFooter.tsx:56]: I think oldDisplayName can be passed without a fallback empty string.

File: src/libs/ReportUtils.ts [hunk 5]
```diff
@@ -2161,11 +2161,11 @@ function getDisplayNameForParticipant(
 
     // If the user's personal details (first name) should be hidden, make sure we return "hidden" instead of the short name
     if (shouldFallbackToHidden && longName === hiddenTranslation) {
-        return longName;
+        return LocalePhoneNumber.formatPhoneNumber(longName);
     }
 
     const shortName = personalDetails.firstName ? personalDetails.firstName : longName;
-    return shouldUseShortForm ? shortName : longName;
+    return LocalePhoneNumber.formatPhoneNumber(shouldUseShortForm ? shortName : longName);
```
Comment (jayeshmangwani) [inline src/libs/ReportUtils.ts:2168]: @Kalydosos I am trying to test this change with various users those have phone numbers, but I’m unsure why we need LocalePhoneNumber.formatPhoneNumber here. Each time I log and check, both shortName and longName are already formatted (no SMS and no country code prefix). 

Could you let me know if there’s a way to test this specific change?

File: src/components/ArchivedReportFooter.tsx (hunk 4 above)
Comment (Kalydosos) [reply src/components/ArchivedReportFooter.tsx:56]: agreed

File: src/libs/ReportUtils.ts (hunk 5 above)
Comment (Kalydosos) [reply src/libs/ReportUtils.ts:2168]: yes they are already formatted at that line
Capture d’écran de 2024-12-06 15-56-26

**Comment by jayeshmangwani** [comment]
@Kalydosos I have tested the changes, and they look good. I was able to see the `.sms` at a few places and noticed differences in phone numbers (with and without country code prefixes). I’m adding the testing steps below. Let me know if these issues are fixable in this PR, or if we should report them and create a new issue:

1. Go to the workspace and invite a user with a phone number. Then, navigate to the `#admins` room for that workspace. You should see the message: `added number@expensify.sms as member` in the chat.
Screenshot 2024-12-09 at 16 38 32

2. Create a new account using a phone number. Then, go to Account -> Contact Method -> Press on the phone number and observe the message.

https://github.com/user-attachments/assets/67414963-8554-4255-b4d4-06e12b9fd363

**Comment by Kalydosos** [comment]
@jayeshmangwani sounds good. For the issues i will give it a quick estimate.

**Comment by jayeshmangwani** [comment]
Thanks 🙌

**Comment by jayeshmangwani** [comment]
Thanks for making changes, @Kalydosos  For issue 1 these are the steps

https://github.com/user-attachments/assets/7f98188d-a373-406d-8bfe-d4a9c39ff3de

**Comment by Kalydosos** [comment]
@jayeshmangwani issue 1 is also fixed. It requires some extra testing.
Capture d’écran de 2024-12-09 18-37-58

**Comment by Kalydosos** [comment]
@jayeshmangwani do you think we could merge soon ? We are having some merging conflicts but anyway testing is priority

**Comment by jayeshmangwani** [comment]
I’m hoping that by tomorrow, I can give a final test and then proceed with the merge

Thanks 🙏

File: src/pages/RoomMemberDetailsPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kalydosos) [reply src/pages/RoomMemberDetailsPage.tsx:51]: @jayeshmangwani I made a test again for this one and i still have the same result., any change on your side for this ?

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by jayeshmangwani — APPROVED** [review]
LGTM 🚀

**Review by luacmartins — APPROVED** [review]
LGTM
